sonarts-sq-plugin/its/plugin/src/test/java/org/sonar/typescript/its/RuleConfigurationTest.java

```diff
@@ -0,0 +1,188 @@
+      {"S4275", "noAccessorFieldMismatchRule"},
+    });
+  }
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    baseDir = temp.newFolder();
+    Tests.runNPMInstall(baseDir, "typescript", "--no-save");
+
+    errorConsumer = new ExternalProcessStreamConsumer();
+    errorConsumer.start();
+    executableBundleFactory = RuleConfigurationTest::createAndDeploy;
+    fileLinesContextFactory = (inputFile) -> mock(FileLinesContext.class);
+    noSonarFilter = mock(NoSonarFilter.class);
+  }
+
+  private static ExecutableBundle createAndDeploy(File deployDestination, Configuration configuration) {
+    try {
+      File file = Tests.PLUGIN_LOCATION.getFile();
+      JarFile jar = new JarFile(file);
+      ZipEntry entry = jar.getEntry("sonarts-bundle.zip");
+      InputStream inputStream = jar.getInputStream(entry);
+      File workDir = temp.newFolder();
+      Zip.extract(inputStream, workDir);
+      return new SonarTSCoreBundle(new File(workDir, "sonarts-bundle"), configuration);
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to extract bundle", e);
+    }
+  }
+
+  @Test
+  public void test() throws Exception {
+    testContext = SensorContextTester.create(baseDir);
+    testContext.fileSystem().setWorkDir(temp.newFolder().toPath());
+
+    List<Integer> expectedLines = prepareTestFile(lintFile);
+    CheckFactory checkFactory = getCheckFactory(ruleKey);
+    ExternalTypescriptSensor sensor = new ExternalTypescriptSensor(executableBundleFactory, noSonarFilter, fileLinesContextFactory, checkFactory, errorConsumer);
+    sensor.execute(testContext);
+
+    assertThat(testContext.allIssues().stream().allMatch(i -> i.ruleKey().rule().equals(ruleKey))).isTrue();
+    List<Integer> actualLines = testContext.allIssues().stream().map(i -> i.primaryLocation().textRange().start().line()).collect(Collectors.toList());
+    assertThat(actualLines).isEqualTo(expectedLines);
+  }
+
+  private List<Integer> prepareTestFile(String tslintRule) throws IOException {
+    Files.write(baseDir.toPath().resolve("tsconfig.json"), "{ \"include\": [\"**/*\"] }".getBytes(StandardCharsets.UTF_8));
+    List<String> lines = Files.readAllLines(lintFile(tslintRule));
+    List<Integer> expectedLines = IntStream.range(0, lines.size())
+      .filter(i -> ERROR_COMMENT.matcher(lines.get(i)).matches())
+      .boxed()
+      .collect(Collectors.toList());
+    testFile(lintFileName(tslintRule), lines.stream().collect(Collectors.joining("\n")));
+    return expectedLines;
+  }
+
+  private Path lintFile(String tslintRule) {
+    return RULE_LINT_FILES.resolve(tslintRule).resolve(lintFileName(tslintRule));
+  }
+
+  private String lintFileName(String tsLintRule) {
+    return tsLintRule + ".lint.ts";
+  }
+
+  private void testFile(String filename, String content) throws IOException {
+    File filePath = new File(baseDir, filename);
+    Files.write(filePath.toPath(), content.getBytes(StandardCharsets.UTF_8));
```

Comment:
@saberduck do we need to write content if it's set through `setContents`?

Reply:
yes, because typescript's `Program` is created from filesystem